homeassistant/components/russound_rio/media_player.py

```diff
@@ -40,22 +31,57 @@ async def async_setup_platform(
 ) -> None:
     """Set up the Russound RIO platform."""
 
-    host = config.get(CONF_HOST)
-    port = config.get(CONF_PORT)
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_IMPORT},
+        data=config,
+    )
+    issue_key = f"deprecated_yaml_{DOMAIN}"
+    translation_key = "deprecated_yaml"
+    if not (
+        result["type"] is FlowResultType.CREATE_ENTRY
+        or result["reason"] == "single_instance_allowed"
+    ):
+        issue_key = f"deprecated_yaml_import_issue_{result['reason']}"
+        translation_key = f"deprecated_yaml_import_issue_{result['reason']}"
+
+    async_create_issue(
+        hass,
+        DOMAIN,
+        issue_key,
+        breaks_in_ha_version="2025.1.0",
+        is_fixable=False,
+        issue_domain=DOMAIN,
+        severity=IssueSeverity.WARNING,
+        translation_key=translation_key,
+        translation_placeholders={
+            "domain": DOMAIN,
+            "integration_title": "Russound RIO",
+        },
+    )
```

Comment:
The homeasssitant domain already has the deprecation issue ready, so we should use that one. checkout the `mpd` `media_player.py` for a flow like that

Reply:
This is functionally identical to the MDP impl. I combined the two async_create_issue calls into one and use the logic above to determine the key.

Reply:
yes and no, there's also a difference in domain, one issue is created using the HOMEASSISTANT_DOMAIN, other onenot

Reply:
Ah, yep I missed that. Will fix.

Reply:
Reverted to the style present in mpd